modules/fbs-core/api/src/main/scala/de/thm/ii/fbs/controller/CheckerConfigurationController.scala

```diff
@@ -153,93 +158,95 @@ class CheckerConfigurationController {
 
   /**
     * Delete a task configuration
-    * @param cid Course id
-    * @param tid Task id
+    *
+    * @param cid  Course id
+    * @param tid  Task id
     * @param ccid Checker configuration id
-    * @param req Http request
-    * @param res Http response
+    * @param req  Http request
+    * @param res  Http response
     */
   @DeleteMapping(value = Array("/{cid}/tasks/{tid}/checker-configurations/{ccid}"))
   def delete(@PathVariable cid: Int, @PathVariable tid: Int, @PathVariable ccid: Int, req: HttpServletRequest, res: HttpServletResponse): Unit = {
     val user = authService.authorize(req, res)
     val privilegedByCourse = crs.getParticipants(cid).find(_.user.id == user.id)
       .exists(p => p.role == CourseRole.DOCENT || p.role == CourseRole.TUTOR)
-
     if (user.globalRole == GlobalRole.ADMIN || user.globalRole == GlobalRole.MODERATOR || privilegedByCourse) {
-      val cc = ccs.getOne(ccid) match {
-        case Some(cc) => cc
+      ccs.getOne(ccid) match {
+        case Some(cc) =>
+          storageService.deleteAllConfigurations(tid, cid, ccid, cc)
         case None => throw new ResourceNotFoundException()
       }
-      if (this.ccs.delete(cid, tid, ccid)) {
-        storageService.deleteConfiguration(ccid)
-        notifyCheckerDelete(tid, cc)
-      }
     } else {
       throw new ForbiddenException()
     }
   }
 
   /**
     * Upload a the main file for a task configuration
-    * @param cid Course id
-    * @param tid Task id
+    *
+    * @param cid  Course id
+    * @param tid  Task id
     * @param ccid Checker configuration id
     * @param file Multipart file
-    * @param req Http request
-    * @param res Http response
+    * @param req  Http request
+    * @param res  Http response
     */
   @PutMapping(value = Array("/{cid}/tasks/{tid}/checker-configurations/{ccid}/main-file"))
   def updateMainFile(@PathVariable cid: Int, @PathVariable tid: Int, @PathVariable ccid: Int,
                      @RequestParam file: MultipartFile,
                      req: HttpServletRequest, res: HttpServletResponse): Unit =
-    uploadFile(storageService.storeMainFile,
+    uploadFile("main-file",
```

Comment:
I would store the name of the `main-file` in as a static member of a global object